desktop/apps/categories3/components/GeneFamilyNav.js

```diff
@@ -41,20 +48,53 @@ const GeneFamilyLink = styled.a`
   }
 `
 
-const GeneFamilyNav = ({ geneFamilies }) => {
-  return (
-    <ResponsiveSidebar>
-      <GeneFamilyList>
-        {geneFamilies.map(geneFamily =>
-          <GeneFamilyItem key={geneFamily.id}>
-            <GeneFamilyLink href={`#${geneFamily.id}`}>
-              {geneFamily.name}
-            </GeneFamilyLink>
-          </GeneFamilyItem>
-        )}
-      </GeneFamilyList>
-    </ResponsiveSidebar>
-  )
+class GeneFamilyNav extends React.Component {
+  constructor (props) {
+    super(props)
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick (e) {
+    e.preventDefault()
+    const id = e.target.hash
+    const section = document.querySelector(id)
+    const topBuffer = 90
+    const scroller = frameAnimator(
+      {
+        duration: 600,
+        startValue: window.scrollY,
+        endValue: section.offsetTop - topBuffer
+      },
+      val => {
+        window.scrollTo(0, val)
+      }
+    )
+    window.requestAnimationFrame(scroller)
```

Comment:
What do you think about the API being something like...

```js
const scroller = frameAnimator(...)
scroller.start()
```

This way it encloses all of its implementation details and the user doesn't need to worry about wiring up?

Reply:
Right now `frameAnimator` is a higher-order function, so it simply returns a function that works nicely with the `window.requestAnimationFrame` API. 

Knowing how to kick that off with `window.requestAnimationFrame(…)`
 is the one thing a caller would need to know. But lemme look into changing the API and see if it makes sense to return an instance of a class instead, thus enabling the interface you're suggesting. 

I'll also extract this to a shared location. Thanks for reviewing @damassi !